query-connector/src/app/backend/user-management.ts

```diff
@@ -105,19 +191,50 @@ export async function updateUserRole(
     }
 
     console.log(`User role updated successfully: ${userId} -> ${newRole}`);
-    return { totalItems: 1, items: [result.rows[0]] };
+    const userWithGroups = await fetchUserGroupMembershipDetails(result);
+
+    return { totalItems: 1, items: userWithGroups };
   } catch (error) {
     console.error("Error updating user role:", error);
     throw error;
   }
 }
 
+/**
+ * Checks whether a given user exists
+ * @param userId - The unique identifier for the user
+ * @returns The given user, if it exists
+ */
+export async function checkUserExists(
+  userId: string,
+): Promise<QCResponse<User>> {
+  if (!userId) {
+    return { totalItems: 0, items: [] };
+  }
+
+  const userCheckQuery = {
+    text: `SELECT * FROM users WHERE id = $1;`,
+    values: [userId],
+  };
+
+  const userCheckResult = await dbClient.query(userCheckQuery);
+  if (userCheckResult.rows.length === 0) {
+    return { totalItems: 0, items: [] };
+  }
+
+  return {
+    totalItems: userCheckResult.rowCount,
+    items: userCheckResult.rows,
+  } as QCResponse<User>;
+}
+
 /**
  * Retrieves all registered users in query connector
  * @returns List of users registered in qc
  */
-export async function getUsers(): Promise<QCResponse<User>> {
-  if (!(await superAdminAccessCheck())) {
+export async function getAllUsers(): Promise<QCResponse<User>> {
```

Comment:
nit: can this be name `getUsers`? it aligns better with the REST naming convention.